test/signs.cc

```diff
@@ -243,6 +260,123 @@ TEST(Signs, TestGetGuideTowardString_roundabout_toward_A1) {
+                     std::make_tuple("Freedom Highway", 0, 0),
+                     std::make_tuple("Valhalla Highway", 0, 0)},
+                    {});
+
+  TryGetGuideString(signs, 0, false, "US 322 West/US 22 West/Freedom Highway/Valhalla Highway");
+  TryGetGuideString(signs, 4, false, "US 322 West/US 22 West/Freedom Highway/Valhalla Highway");
+  TryGetGuideString(signs, 3, false, "US 322 West/US 22 West/Freedom Highway");
+  TryGetGuideString(signs, 2, false, "US 322 West/US 22 West");
+  TryGetGuideString(signs, 1, false, "US 322 West");
+
+  TryGetGuideString(signs, 0, true, "US 322 West");
+  TryGetGuideString(signs, 4, true, "US 322 West");
+  TryGetGuideString(signs, 2, true, "US 322 West");
+  TryGetGuideString(signs, 1, true, "US 322 West");
+}
+
+TEST(Signs, TestGetGuideString_TowardOnly) {
+  // Create guide sign
+  // Specify input in descending consecutive count order
+  Signs signs = GetGuideSigns({}, {std::make_tuple("A 1", 1, 1), std::make_tuple("Remscheid", 0, 1),
+                                   std::make_tuple("Wermelskirchen", 0, 0),
+                                   std::make_tuple("Hückeswagen", 0, 0)});
+
+  TryGetGuideString(signs, 0, false, "A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 4, false, "A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 3, false, "A 1/Remscheid/Wermelskirchen");
+  TryGetGuideString(signs, 2, false, "A 1/Remscheid");
+  TryGetGuideString(signs, 1, false, "A 1");
+
+  TryGetGuideString(signs, 0, true, "A 1/Remscheid");
+  TryGetGuideString(signs, 4, true, "A 1/Remscheid");
+  TryGetGuideString(signs, 2, true, "A 1/Remscheid");
+  TryGetGuideString(signs, 1, true, "A 1");
+}
+
+TEST(Signs, TestGetGuideString_NoConsecutiveCount) {
+  // Create guide sign
+  // Specify input in descending consecutive count order
+  Signs signs =
+      GetGuideSigns({std::make_tuple("US 322 West", 1, 0), std::make_tuple("US 22 West", 1, 0),
+                     std::make_tuple("Freedom Highway", 0, 0),
+                     std::make_tuple("Valhalla Highway", 0, 0)},
+                    {std::make_tuple("A 1", 1, 0), std::make_tuple("Remscheid", 0, 0),
+                     std::make_tuple("Wermelskirchen", 0, 0), std::make_tuple("Hückeswagen", 0, 0)});
+
+  TryGetGuideString(
+      signs, 0, false,
+      "US 322 West/US 22 West/Freedom Highway/Valhalla Highway/A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 4, false, "US 322 West/US 22 West/A 1/Remscheid");
+  TryGetGuideString(signs, 3, false, "US 322 West/US 22 West/A 1");
+  TryGetGuideString(signs, 2, false, "US 322 West/A 1");
+  TryGetGuideString(signs, 1, false, "US 322 West");
+
+  TryGetGuideString(
+      signs, 0, true,
+      "US 322 West/US 22 West/Freedom Highway/Valhalla Highway/A 1/Remscheid/Wermelskirchen/Hückeswagen");
+  TryGetGuideString(signs, 4, true, "US 322 West/US 22 West/A 1/Remscheid");
```

Comment:
Why is `Remscheid` being returned here? I thought if `limit_by_consecutive_count` is set, only entries with the same consecutive count are returned? In this case `A 1` has a consecutive count of 1, while `Remscheid`'s is 0, which should make us drop `Remscheid`, no?

Reply:
> Why is `Remscheid` being returned here? I thought if `limit_by_consecutive_count` is set, only entries with the same consecutive count are returned? In this case `A 1` has a consecutive count of 1, while `Remscheid`'s is 0, which should make us drop `Remscheid`, no?

Sorry, ignore this question. Thats not consecutive count I was looking at, but the is route attribute.